lib/actions/deleteObject.js

```diff
@@ -1,31 +1,94 @@
+*/
+module.exports.getMetaModel = async function getMetaModel(configuration) {
+  configuration.metaType = 'lookup';
   const metaLoader = new MetaLoader(configuration, this);
-  return metaLoader.getObjectTypes();
+  const metaData = await metaLoader.loadMetadata();
+  metaData.in.properties[configuration.lookupField].required = true;
+  return metaData;
 };
 
+/**
+ * Des: 
+ * - This function will firstly lookup the object based on fields
+ * in the metaData
+ * - Emits an error if zero or more than one object is found and returns false
+ * - If delete was sucessful
+ * @implementedBy C.J.V
+ * @param configuration encapsulates the lookup value and lookup key
+*/
+module.exports.process = async function process(message, configuration) {
+  const { lookupField } = configuration;
+  const lookupValue = message.body[lookupField];
+  const res = [];
+  const sfConn = sfConnection.createConnection(configuration, this);
 
-exports.process = async function deleteObject(message, configuration) {
-  if (!message.body.id || !String(message.body.id).trim()) {
-    this.logger.error('Salesforce error. Empty ID');
-    return messages.newMessageWithBody({ });
+  if (!lookupValue) {
+    const err = new Error('No unique criteria provided');
+    this.logger.error(err);
+    this.emit('error', err);
+    return;
   }
 
   this.logger.debug(`Preparing to delete a ${configuration.sobject} object...`);
 
-  let response;
-  try {
-    const sfConn = sfConnection.createConnection(configuration, this);
-    response = await sfConn.sobject(configuration.sobject).delete(message.body.id);
-  } catch (err) {
-    this.logger.error(`Salesforce error. ${err.message}`);
-    return messages.newMessageWithBody({ });
-  }
+  const meta = await sfConn.describe(configuration.sobject);
+  const field = meta.fields.find(fld => fld.name === lookupField);
+  this.logger.info(MetaLoader.TYPES_MAP[field.type]);
+  const condition = (['date','datetime'].includes(field.type) || MetaLoader.TYPES_MAP[field.type] === 'number')
+    ? `${lookupField} = ${lookupValue}`
+    : `${lookupField} = '${lookupValue}'`;
+
+  await sfConn.sobject(configuration.sobject)
+    .select('*')
+    .where(condition)
+    .on('record', (record) => {
+      res.push(record);
+    })
+    .on('end', async () => {
+      if (res.length === 0) {
+
+        this.logger.info('No objects are found');
+        this.emit('data', messages.newMessageWithBody({}));
+
+      } else if (res.length === 1) {
 
-  this.logger.debug(`${configuration.sobject} has been successfully deleted (ID = ${response.id}).`);
-  return messages.newMessageWithBody({ result: response });
+        await helpers.deleteObjById.call(this, sfConn, res[0].Id, configuration.sobject);
+       
+      } else {
+
+        const err = new Error('More than one object found, can only delete 1');
+        this.logger.error(err);
+        this.logger.info(`Here are the objects found ${JSON.stringify(res)}`);
```

Comment:
```suggestion
        this.logger.trace(`Here are the objects found ${JSON.stringify(res)}`);
```

Reply:
Use trace level when logging js objects. Info is default logging level on our platform. JS objects can generate a lot of unnecessary visual flood.

Reply:
Trace level was implemented where required